pkg/controller/node/nodecontroller.go

```diff
@@ -375,7 +375,7 @@ func (nc *NodeController) allocateOrOccupyCIDR(obj interface{}) {
 	glog.V(4).Infof("Assigning node %s CIDR %s", node.Name, podCIDR)
 	for rep := 0; rep < podCIDRUpdateRetry; rep++ {
 		node.Spec.PodCIDR = podCIDR.String()
-		if _, err := nc.kubeClient.Core().Nodes().Update(node); err == nil {
+		if _, err := nc.kubeClient.Core().Nodes().Update(node); err != nil {
 			glog.Errorf("Failed while updating Node.Spec.PodCIDR : %v", err)
```

Comment:
Can you make it clearer that this isn't an actual failure but one that we might retry? i.e just will retry %d times, (podCIDRUpdateRetry-rep).